src/libaktualizr/crypto/keymanager.cc

```diff
@@ -189,13 +189,13 @@ std::string KeyManager::getCN() const {
   return cn;
 }
 
-void KeyManager::copyCertsToCurl(const std::shared_ptr<HttpInterface> &http) {
+void KeyManager::copyCertsToCurl(HttpInterface &http) {
   std::string pkey = getPkey();
   std::string cert = getCert();
   std::string ca = getCa();
 
   if ((pkey.size() != 0u) && (cert.size() != 0u) && (ca.size() != 0u)) {
-    http->setCerts(ca, config_.tls_ca_source, cert, config_.tls_cert_source, pkey, config_.tls_pkey_source);
+    http.setCerts(ca, config_.tls_ca_source, cert, config_.tls_cert_source, pkey, config_.tls_pkey_source);
```

Comment:
I'm not sure I understand the benefit here of the reference instead of the shared_ptr. How is that actually more flexible?

Reply:
In that I (the caller) can have plain HttpInterface pointer and still use this interface.

Reply:
There seems to be no reason to use <shared_ptr> here, because `copyCertsToCurl` doesn't share or store this pointer anywhere, it just uses it and forgets about it after it returns.

Reply:
Yes, there is no dealing with ownership here, a reference is the obvious choice.

See this nice blogpost (again, sorry :)): https://herbsutter.com/2013/06/05/gotw-91-solution-smart-pointer-parameters/

Reply:
Ah, I see now. Yes, agreed. And the blog post is excellent, I wish that type of logic was more generally observed and enforced. (I mean in general; I think we mostly do it right.)